mico-core/src/main/java/io/github/ust/mico/core/broker/DeploymentBroker.java

```diff
@@ -79,13 +79,15 @@ public MicoApplicationJobStatus deployApplication(String shortName, String versi
                 .setStatus(MicoServiceBackgroundJob.Status.RUNNING);
             backgroundJobBroker.saveJob(job);
 
-            Optional<MicoServiceDeploymentInfo> serviceDeploymentInfoOptional = serviceDeploymentInfoRepository
+            List<MicoServiceDeploymentInfo> serviceDeploymentInfos = serviceDeploymentInfoRepository
                 .findByApplicationAndService(micoApplication.getShortName(), micoApplication.getVersion(),
                     micoService.getShortName(), micoService.getVersion());
-            MicoServiceDeploymentInfo serviceDeploymentInfo = serviceDeploymentInfoOptional.orElseThrow(() ->
-                new IllegalStateException("Service deployment information for service '" + micoService.getShortName()
+            if(serviceDeploymentInfos.isEmpty()) {
```

Comment:
Maybe also throw error for the undefined case of more than one serviceDeploymentInfo for a normal service. Ordering in the list may not be deterministic.